### PR TITLE
[Drone] Add rips to drone on 4.0-dev

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

We have a new partnership for a static code analysis tool as mentioned here:
https://developer.joomla.org/news/739-rips-becomes-joomla-official-code-analysis-partner.html

Add rips to drone on 4.0-dev based on (https://github.com/joomla/joomla-cms/pull/20796)

This PR adds the tool to our CI pipeline.

### Testing Instructions
Trigger a build in drone.

### Expected result
Works.

cc @SniperSister 

@wilsonge please await final confirmation before merging thanks.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -66,6 +66,15 @@ pipeline:
       - fluxbox  > /dev/null 2>&1 &
       - node_modules/karma/bin/karma start node_modules/joomla-javascript-tests/src/karma.conf.js --single-run
 
+  analysis:
+    image: rips/rips-cli
+    secrets:  [rips_username, rips_password]
```
Comment (Quy) [inline .drone.yml]: Remove extra space?

**Comment by rdeutz** [comment]
I would make it the last step after system tests

**Comment by zero-24** [comment]
@rdeutz @wilsonge Please review & merge as this now enables the drone security checks on 4.0-dev too (this also fixes an code style issue that cause drone to fail now)

**Comment by wilsonge** [comment]
Leaving this for @rdeutz to approve

**Comment by zero-24** [comment]
This has now been integrated here: https://github.com/joomla/joomla-cms/pull/21853 so we don't split up the different configs in the branches we use.

**Pull request closed without merging**